src/go/supervisor/pkg/svc/service.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright © 2021 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package svc
+
+import (
+	"fmt"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+	"golang.org/x/sys/windows/svc/eventlog"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/supervisor/pkg/port"
+)
+
+const (
+	portSrvrAddr = "127.0.0.1"
+	portSrvrPort = 4444
+)
+
+func RunService(name string, isDebug bool) {
+	elog := initElogger(name, isDebug)
+	elog.Info(uint32(windows.NO_ERROR), fmt.Sprintf("starting %s service", name))
+	run := svc.Run
+	if isDebug {
+		run = debug.Run
+	}
+
+	portServer := port.NewServer(portSrvrAddr, portSrvrPort, elog)
+	supervisor := NewSupervisor(portServer, elog)
+	err := run(name, supervisor)
+	if err != nil {
+		elog.Error(uint32(windows.ERROR_EXCEPTION_IN_SERVICE), fmt.Sprintf("%s service failed: %v", name, err))
+		return
+	}
+	elog.Info(uint32(windows.NO_ERROR), fmt.Sprintf("%s service stopped", name))
+}
+
+// TODO: maybe create a combined logger to log both logrus and elog
+func initElogger(name string, isDebug bool) debug.Log {
+	var elog debug.Log
+	var err error
+	if isDebug {
+		elog = debug.New(name)
+	} else {
+		elog, err = eventlog.Open(name)
+		if err != nil {
+			return elog
+		}
+	}
+	defer elog.Close()
```

Comment:
Close the log at the end of the function, and then end the function (immediately closing the log)‽